fix: Implement Caching for Model Responses and Models List

syuzanvs opened the pull request:
- Added the `checkModel` method to improve error handling and caching of AI models.
- Added a check to ensure that the models are fetched and cached correctly from the API.
- Enhanced user feedback by displaying available models when an invalid model is specified.
- Ensured that the caching mechanism properly handles expired models.

Closes #113

syuzanvs committed:
fix: Implement Caching for Model Responses and Models List

- Added the `checkModel` method to improve error handling and caching of AI models.
- Added a check to ensure that the models are fetched and cached correctly from the API.
- Enhanced user feedback by displaying available models when an invalid model is specified.
- Ensured that the caching mechanism properly handles expired models.

Closes #113

jjroelofs committed:
Merge branch '1.x' into syuzan/1.x/#113-implement-caching-for-model-responses-and-models-list

jjroelofs reviewed (changes requested):
Excellent work, it works great, I just want to ask you to move the stored data to the ckeditor namespace in localStorage to prevent potential conflicts with other software. Since this is a small fix I will probably do it myself.

Commit:
feat(storage): implement model cache with ck5-ai-agent namespace

- Add STORAGE_PREFIX constant for localStorage namespace
- Store model lists under ck5-ai-agent:{engine}_models
- Cache model lists for 24 hours
- Extract cacheModels into separate method
